test/functional/wallet_musig.py

```diff
@@ -28,6 +28,178 @@ def set_test_params(self):
+
+        # Only 2 out of 3 provide partial sigs
+        psig_psbts = []
+        for i in range(2):
+            proc = wallets[i].walletprocesspsbt(psbt=comb_nonce_psbt)
+            psig_psbts.append(proc["psbt"])
+
+        comb_psig_psbt = self.nodes[0].combinepsbt(psig_psbts)
+
+        # Finalization fails due to missing partial sig
+        finalized = self.nodes[0].finalizepsbt(comb_psig_psbt)
+        assert_equal(finalized["complete"], False)
+
+        # Still only two partial sigs in combined PSBT
+        dec = self.nodes[0].decodepsbt(comb_psig_psbt)
+        assert_equal(len(dec["inputs"][0]["musig2_partial_sigs"]), 2)
+
+        self.log.info("Testing mismatched key order")
+        wallets = []
+        keys = []
+        for i in range(2):
+            wallet_name = f"musig_{self.WALLET_NUM}"
+            self.WALLET_NUM += 1
+            self.nodes[0].createwallet(wallet_name)
+            wallet = self.nodes[0].get_wallet_rpc(wallet_name)
+            wallets.append(wallet)
+
+            for priv_desc in wallet.listdescriptors(True)["descriptors"]:
+                desc = priv_desc["desc"]
+                if not desc.startswith("tr("):
+                    continue
+                privkey = PRIVKEY_RE.search(desc).group(1)
+                break
+            for pub_desc in wallet.listdescriptors()["descriptors"]:
+                desc = pub_desc["desc"]
+                if not desc.startswith("tr("):
+                    continue
+                pubkey = PUBKEY_RE.search(desc).group(1)
+                privkey += ORIGIN_PATH_RE.search(pubkey).group(1)
+                break
+            keys.append((privkey, pubkey))
+
+        desc0 = f"rawtr(musig({keys[0][0]},{keys[1][1]}))"
+        wallets[0].importdescriptors([{
+            "desc": descsum_create(desc0),
+            "active": True,
+            "timestamp": "now",
+        }])
+
+        # Reverse order to desc0
+        desc1 = f"rawtr(musig({keys[1][0]},{keys[0][1]}))"
+        wallets[1].importdescriptors([{
+            "desc": descsum_create(desc1),
+            "active": True,
+            "timestamp": "now",
+        }])
+
+        # Addresses should be different due to different key ordering
+        addr0 = wallets[0].getnewaddress(address_type="bech32m")
+        addr1 = wallets[1].getnewaddress(address_type="bech32m")
+        assert addr0 != addr1
+
+        self.log.info("Testing finalize without partial sigs")
+        wallets, psbt = self.setup_musig_scenario(num_wallets=2)
+        nonce_psbts = [w.walletprocesspsbt(psbt=psbt)["psbt"] for w in wallets]
+        comb_nonce_psbt = self.nodes[0].combinepsbt(nonce_psbts)
+
+        finalized = self.nodes[0].finalizepsbt(comb_nonce_psbt)
+        assert_equal(finalized["complete"], False)
+
+        dec = self.nodes[0].decodepsbt(comb_nonce_psbt)
+        assert "musig2_pubnonces" in dec["inputs"][0]
+        assert "musig2_partial_sigs" not in dec["inputs"][0]
+
+
```

Comment:
In c7c9dc2d8634bc79e76783be113f8d42cee63c6e "test: Add musig failure scenarios"

Nit: extra blank line